signac/core/collection_mongodb.py

```diff
@@ -35,10 +34,7 @@ def _sync(self):
         """Write the data from MongoDB."""
         data = self.to_base()
         data_to_insert = {self._key: self._name, 'data': data}
-        try:
-            self._collection.replace_one({self._key: self._name}, data_to_insert, True)
-        except self._errors.InvalidDocument as err:
-            raise TypeError(str(err))
```

Comment:
Was the try-except just performing the JSON validation before?

Reply:
This was changing the exception type (`InvalidDocument` to `TypeError`).  Now, this exception is raised by validator .

Reply:
@vyasr What should we done regarding this? Should I revert the change? (as this was removed because I added `JSONFormatValidator` in `MongoDBCollection`)